### PR TITLE
[SYCL] Windows MS-Compatibility changes

### PR DESCRIPTION
**Pull request by premanandrao**
Signed-off-by: Premanand M Rao <premanand.m.rao@intel.com>

This is a continuation of: intel/llvm#446

### CODE REVIEW

File: clang/test/SemaSYCL/delayed-parsing2.cpp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// RUN: %clang %s -fsycl -c
+// REQUIRES: system-windows
+
+#include <windows.h>
```
Comment (erichkeane) [inline clang/test/SemaSYCL/delayed-parsing2.cpp]: Generally you shouldn't use #includes of system headers in lit tests, it makes them fragile, unportable, and non-deterministic.  Please test only what you mean.

File: clang/test/SemaSYCL/delayed-parsing.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s
+// RUN: %clang_cc1 -std=c++1y -verify -fsyntax-only %s -DDELAYED_TEMPLATE_PARSING -fdelayed-template-parsing
+// RUN: %clang %s -DDELAYED_TEMPLATE_PARSING -fsycl -c
+// REQUIRES: system-windows
```
Comment (erichkeane) [inline clang/test/SemaSYCL/delayed-parsing.cpp]: Why is this required?  This test shouldn't depend on the platform.

File: clang/test/Preprocessor/sycl-macro2.cpp [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// RUN: %clang -c %s -fsycl -E -dM -o - | FileCheck --check-prefix=CHECK-MSVC %s
+// REQUIRES: system-windows
```
Comment (erichkeane) [inline clang/test/Preprocessor/sycl-macro2.cpp]: See below.  Use triples if you care about platform.

File: clang/test/SemaSYCL/delayed-parsing2.cpp (hunk 1 above)
Comment (mibintc) [reply clang/test/SemaSYCL/delayed-parsing2.cpp]: There's another place in the tree where system includes are used, maybe basic_tests or unittests?

Comment (bader) [reply clang/test/SemaSYCL/delayed-parsing2.cpp]: Do we really need this test?
I think clang/test/Preprocessor/sycl-macro.cpp fully covers the fix.

Comment (premanandrao) [reply clang/test/SemaSYCL/delayed-parsing2.cpp]: @bader, if you believe this test is redundant with sycl-macro.cpp, I will remove it.  Thanks.

File: clang/test/Preprocessor/sycl-macro2.cpp (hunk 3 above)
Comment (premanandrao) [reply clang/test/Preprocessor/sycl-macro2.cpp]: @erichkeane, thanks I will just remove this test.   Alexey feels it is redundant.

File: clang/test/SemaSYCL/delayed-parsing.cpp (hunk 2 above)
Comment (premanandrao) [reply clang/test/SemaSYCL/delayed-parsing.cpp]: So what I was trying to do is:
1.  create a test case that needs -fdelayed-parsing to compile cleanly
2.  demo that with -fdelayed-parsing it does compile cleanly
These are the first two RUN cases.   These are not platform-dependent
3.  was to demo that the clang driver passes the required options to the compiler and compiles it cleanly.
For this, I needed the platform dependency, since the Linux compiler does not need the delayed-parsing option.  I could not figure out how to use triples with the clang driver.

Is there a platform-neutral way to demo 3?  I could use that instead.  Thanks.

File: clang/test/SemaSYCL/delayed-parsing2.cpp (hunk 1 above)
Comment (bader) [reply clang/test/SemaSYCL/delayed-parsing2.cpp]: I agree with @erichkeane, that this test doesn't belong to clang's lit regression tests.

Probably the proper test should be a driver test checking that options are passed to the device compiler when we compile for Windows OS.
